### PR TITLE
Converted the plugin to a caddy2 plugin

### PR DESCRIPTION
**Pull request by tdralle**
This pull request actually replaces the current caddy1 plugin by a newer version targeting caddy2. 

Since the plugin model of caddy1 and caddy2 are incompatible this commit will actually introduce a "breaking change" for those which still rely on caddy1. However, caddy2 is out for quite some time now i think its safe to bump it here. The old versions are still tagged and can be accessed if needed.

### CODE REVIEW

**Review by mholt — APPROVED** [review]
Oh this is good :) I just did a quick scan. I'll see if another project maintainer wants to look this over before merging.

**Comment by tdralle** [comment]
Thanks for the thumbs up so far. Looking forward for the other maintainers to share their opinion. 

One hint, once this pull request is accepted you can also close https://github.com/restic/caddy/issues/2. The append only option can now be configured in the json configuration file straight.

**Comment by mholt** [comment]
Thank you!!

**Pull request merged**
